controller/appcontroller.go

```diff
@@ -840,6 +858,12 @@ func (ctrl *ApplicationController) Run(ctx context.Context, statusProcessors int
 func (ctrl *ApplicationController) requestAppRefresh(appName string, compareWith *CompareWith, after *time.Duration) {
 	key := ctrl.toAppKey(appName)
 
+	obj, exists, err := ctrl.appInformer.GetIndexer().GetByKey(key)
+	app, ok := obj.(*appv1.Application)
+	if exists && err == nil && ok {
+		ctrl.metricsServer.IncRefresh(app, compareWith.String())
```

Comment:
I am wondering what additional value this will add?

The controller has metrics (https://github.com/argoproj/argo-cd/pull/17013) on the refreshQueue that will give the same information as this metrics.

Additionnaly, the refresh/[reconciliation metric](https://github.com/argoproj/argo-cd/blob/7e80f1e8e59da2701c87e97b6bda4c64345a90e2/controller/appcontroller.go#L1484) is already provided.

Maybe adding the compare_with to the existing metrics would help?

Reply:
Based on the code path, this will be almost the same as the reconciliation. Reconciliation will be increased less often than your new metric because a lot of refreshRequest that does not require a reconciliation will be discarded.

Afaik, refreshRequest that does not require a reconciliation (reconciliation here is what happens when you click refresh in the ui, when an app resources has changed, when timeout expires, when orphan resource in the same namespace changes, etc) have a very low impact on performance.

Reply:
Hi! After apply your fix, honestly I think this PR doesnt make sense and can be closed.